source/server/feature_toggles.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+namespace nidevice_grpc {
+
+using FeatureToggleConfigurationMap = std::unordered_map<std::string, bool>;
+class FeatureToggles {
+ public:
+  FeatureToggles() {}
+  FeatureToggles(FeatureToggleConfigurationMap&& map) : map_(map) {}
```

Comment:
I don't know if I've used this before. Is the rvlaue reference here an optimization to avoid unneccessary copies?

Reply:
Yeah. It goes with the `std::move` in the parser and then passes through to the move constructor of `std::map`. Basically the logic is that `FeatureToggles` will always "own" the map even though it doesn't construct it. I think there's a pretty decent chance that it would get optimized to the same thing if it was passed by value.

Reply:
Cool, I wanted to make sure I was understanding it right. That sounds good to me, matches the standard use case I found mentioned on stackoverflow and such.